src/content/docs/developer-guide/git-guide.md

```diff
@@ -0,0 +1,178 @@
+
+```sh
+git add src/content/docs/developer-guide/picture.png
+git commit -m “Added picture.png”
+```
+
+As you can see, you do not have to mention that this file has to be added to LFS. `git` will automatically detect the file extension, and if it is tracked by LFS, it will be handled by LFS.
+
+## How to sign your commits
+
+With git, you can use any username and email address you want to make a commit, even a name and email that does not belong to you! In small teams, or closed projects, this is not an issue. However, in open source work, it can create confusion.
+
+Thankfully, there are several ways to sign your commits and confirm your identity. We require commits to be signed for our repo, because it helps us improve the security of the project.
+
+To find out more about how to sign your commits, check out the [official GitHub documentation on the topic](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
+
+## Skipping the pre-commit hooks
+
+We use pre-commit and commit hooks to run linting, tests, and other types of checks. To skip these tests, you can use the `--no-verify` flag, or its shorthand `-n`.
+
+```sh
+git commit --no-verify -m “message”
+```
+
+The `--no-verify` flag is also available for the `git push` command, and it serves the same function. 
+
+However, it's important to keep in mind that these tests are only skipped locally, and the flag is generally used to speed up development. Once you make a PR, you have to ensure that all the tests complete successfully.
+
+## Good practices
+
+### git rebase
+
+As you contribute to open source projects, you may run into conflicts between your PR or branch and the upstream code. This usually happens when your commit is made on an older version of the code base, and new changes have been implemented by other contributors.
+
+With the `git rebase` command you can move your branch along the history of the repo and change the commit it is based on. This will allow you to base your branch on the latest commit. The command for this is:
+
+```sh
+git rebase origin/main
+```
+
+It's important to note that `origin/main` has to be up to date with `ratatui/main`. This can be done in a number of ways, including by [setting `ratatui/main` as the upstream for your repo](#upstream), or through the GitHub interface.
+
+![Updating a branch on Github](./update.png)
+
+Rebasing is very straightforward in theory. However, you may run into conflicts that have to be resolved before you can rebase your branch. One of the main causes for rebase conflicts is the presence of many small commits in your branch history. This is where squashing commits is useful.
+
```

Comment:
A sub-section `Squashing` here could be nice to help new people differentiate between rebase and squash. Especially since it uses the same `rebase` command.
Plus we will be able to reference this in links later to guide people.